src/theory/fp/theory_fp.cpp

```diff
@@ -866,7 +866,7 @@ bool TheoryFp::collectModelValues(TheoryModel* m,
 
       // At most one of the flags (NaN, inf, zero) can be set
       Node one = nm->mkConst(BitVector(1U, 1U));
-      size_t numFlags = 0;
+      size_t numFlags CVC5_UNUSED = 0;
```

Comment:
If `ee->getRepresentative(compNaN)` et al. can be stored in a Node (which, really, it should be able to be) this could be simplified to:
```
Assert((cNaN == one ? 1 : 0) + (cInf == one ? 1: 0) + (cZero == one ? 1 : 0) <= 1);
```

so avoiding the need to compute any of this if it is not being checked.

Reply:
Ah, good point. I have refactored things a bit.